resources/fetching/fetchers/azure/assets_fetcher_test.go

```diff
@@ -0,0 +1,121 @@
+	"context"
+	"testing"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+	"github.com/elastic/cloudbeat/resources/utils/testhelper"
+)
+
+type AzureAssetsFetcherTestSuite struct {
+	suite.Suite
+
+	resourceCh chan fetching.ResourceInfo
+}
+
+func TestAzureAssetsFetcherTestSuite(t *testing.T) {
+	s := new(AzureAssetsFetcherTestSuite)
+
+	suite.Run(t, s)
+}
+
+func (s *AzureAssetsFetcherTestSuite) SetupTest() {
+	s.resourceCh = make(chan fetching.ResourceInfo, 50)
+}
+
+func (s *AzureAssetsFetcherTestSuite) TearDownTest() {
+	close(s.resourceCh)
+}
+
+func (s *AzureAssetsFetcherTestSuite) TestFetcher_Fetch() {
+	ctx := context.Background()
+	mockInventoryService := &inventory.MockServiceAPI{}
+	fetcher := AzureAssetsFetcher{
+		log:        testhelper.NewLogger(s.T()),
+		resourceCh: s.resourceCh,
+		provider:   mockInventoryService,
+	}
+
+	mockAssets := []inventory.AzureAsset{
+		{
+			Id:             "id1",
+			Name:           "name1",
+			Location:       "location1",
+			Properties:     map[string]interface{}{"key1": "value1"},
+			ResourceGroup:  "rg1",
+			SubscriptionId: "subId1",
+			TenantId:       "tenantId1",
+			Type:           inventory.VirtualMachineAssetType,
+		},
+		{
+			Id:             "id2",
+			Name:           "name2",
+			Location:       "location2",
+			Properties:     map[string]interface{}{"key2": "value2"},
+			ResourceGroup:  "rg2",
+			SubscriptionId: "subId2",
+			TenantId:       "tenantId2",
+			Type:           inventory.StorageAccountAssetType,
+		},
+	}
+
+	mockInventoryService.On("ListAllAssetTypesByName", mock.MatchedBy(func(assets []string) bool {
+		return true
+	})).Return(
+		mockAssets, nil,
+	)
+
+	err := fetcher.Fetch(ctx, fetching.CycleMetadata{})
+	s.NoError(err)
+	results := testhelper.CollectResources(s.resourceCh)
+
+	s.Equal(len(AzureResourceTypes), len(results))
+
+	lo.ForEach(results, func(r fetching.ResourceInfo, index int) {
+		data := r.GetData()
+		s.NotNil(data)
+		resource := data.(inventory.AzureAsset)
+		s.NotEmpty(resource)
+		s.Equal(mockAssets[index].Id, resource.Id)
+		s.Equal(mockAssets[index].Name, resource.Name)
+		s.Equal(mockAssets[index].Location, resource.Location)
+		s.Equal(mockAssets[index].Properties, resource.Properties)
+		s.Equal(mockAssets[index].ResourceGroup, resource.ResourceGroup)
+		s.Equal(mockAssets[index].SubscriptionId, resource.SubscriptionId)
+		s.Equal(mockAssets[index].TenantId, resource.TenantId)
+		s.Equal(mockAssets[index].Type, resource.Type)
+		meta, err := r.GetMetadata()
+		s.NoError(err)
+		s.NotNil(meta)
+		s.NoError(err)
+		s.NotEmpty(meta)
+		s.Equal(mockAssets[index].Id, meta.ID)
+		s.Equal(AzureResourceTypes[mockAssets[index].Type], meta.Type)
+		s.Equal("", meta.SubType)
+		s.Equal(mockAssets[index].Name, meta.Name)
+		s.Equal(mockAssets[index].Location, meta.Region)
```

Comment:
```suggestion
		s.Equal(mockAssets[index].Id, meta.ID)
		s.Equal(AzureResourceTypes[mockAssets[index].Type], meta.Type)
		s.Equal("", meta.SubType)
		s.Equal(mockAssets[index].Name, meta.Name)
		s.Equal(mockAssets[index].Location, meta.Region)
```

I would construct an expected `meta` object and compare it using the `Equal` function, this produces a better diff and it is more feature-proof as we can catch cases where we forget to use new field if we combine it with `exhauststruct`.